data/builtins/thingengine.builtin/dataset.tt.in

```diff
@@ -138,4 +138,15 @@ dataset @org.thingpedia.builtin.thingengine.builtin language "en" {
     program (p_unit : Enum(celsius,fahrenheit,kelvin)) = @org.thingpedia.builtin.thingengine.builtin.set_temperature_unit(unit=p_unit)
     #_[utterances=["use ${p_unit} for temperatures", "use ${p_unit} to display temperatures",
                    "tell me the temperature in ${p_unit}"]];
+
+    program(p_date : Date, p_message : String) = ontimer(date=[p_date]) => @org.thingpedia.builtin.thingengine.builtin.say(message=p_message)
+    #_[utterances=["remind me to ${p_message} at ${p_time}",
+                    "tell me to ${p_message} at ${p_time}",
+                    "set an alarm for ${p_time}"]];
+
+    program(p_num : Number, p_unit : Enum(hour, minute, second)) = ontimer(date=[new Date()]) => @org.thingpedia.builtin.thingengine.builtin.say(message="Time's up!")
+    #_[utterances=["set a timer for ${p_num} ${p_unit}"]];
```

Comment:
This is not correct, if any, it should be `p_duration : Measure(ms)`, and then it should be `ontimer(date=[$now + p_duration]) => ...`

The other thing, I don't think we can hardcode a string in the primitive template. Constants in a ThingTalk program must be mentioned in the command, one way or another. So we'll need another builtin function that plays the alert sound and says "Time's up!".